AuthenticatorShared/Core/Vault/Repositories/AuthenticatorItemRepository.swift

```diff
@@ -89,35 +90,112 @@ protocol AuthenticatorItemRepository: AnyObject {
+    ///   - configService: Service to determine if the sync feature flag is turned on.
+    ///   - cryptographyService: Service to encrypt/decrypt locally stored Authenticator items.
+    ///   - sharedItemService: Service to fetch items from the shared CoreData store - shared from
+    ///     the main Bitwarden PM app.
+    ///   - errorReporter: Error Reporter for any errors encountered
+    ///   - timeProvider: A protocol wrapping the present time.
+    ///   - totpService: A service for refreshing TOTP codes.
     init(
         authenticatorItemService: AuthenticatorItemService,
+        configService: ConfigService,
         cryptographyService: CryptographyService,
         errorReporter: ErrorReporter,
+        sharedItemService: AuthenticatorBridgeItemService,
         timeProvider: TimeProvider,
         totpService: TOTPService
     ) {
         self.authenticatorItemService = authenticatorItemService
+        self.configService = configService
         self.cryptographyService = cryptographyService
         self.errorReporter = errorReporter
         self.timeProvider = timeProvider
         self.totpService = totpService
+        self.sharedItemService = sharedItemService
     }
 
     // MARK: Private Methods
 
+    /// Combine sections that are locally stored with the list of the sections created with the shared items,
+    /// when sync with the PM app is enabled.
+    ///
+    /// Note: If the `enablePasswordManagerSync` feature flag is not enabled, or if the user has not yet
+    /// turned on sync for any accounts, this method simply returns `localSections`.
+    ///
+    /// - Parameters:
+    ///   - localSections: The [ItemListSection] sections for the items locally stored
+    ///   - sharedItems: The shared items that are coming in via sync with the PM app
+    /// - Returns: A list of the sections to display in the item list
+    ///
+    private func combinedSections(
+        localSections: [ItemListSection],
+        sharedItems: [AuthenticatorBridgeItemDataView]
+    ) async throws -> [ItemListSection] {
+        guard await configService.getFeatureFlag(.enablePasswordManagerSync),
+              await sharedItemService.isSyncOn() else {
+            return localSections
+        }
+
+        let groupedByAccount = Dictionary(
+            grouping: sharedItems,
+            by: { item in
+                if let email = item.accountEmail, !email.isEmpty {
+                    if let domain = item.accountDomain, !domain.isEmpty {
+                        return email + " | " + domain
+                    } else {
+                        return email
+                    }
+                } else if let domain = item.accountDomain, !domain.isEmpty {
+                    return domain
+                } else {
+                    return ""
+                }
+            }
+        )
+
+        var sections = localSections
```

Comment:
⚠️ In order to match the previous behavior you need to filter the empty items here. Not sure if it's possible for local sections but feels like a good thing to add.
```suggestion
        var sections = localSections.filter { !$0.items.isEmpty }
```

Reply:
Those should already be filtered, since `localSections` is the result of calling the original `itemListSections` func and the last line of that is still:
``` Swift
        .filter { !$0.items.isEmpty }
```

When I added the last round of guards I considered keeping in the filtering just in case the original method somehow changed. But that felt like beyond the job of this method. This method is only supposed to be adding in the shared sections and assuming the original sections are already filtered seemed like the right balance.

But let me know if you think differently. 👍

Reply:
Makes perfect sense thanks!